controllers/analytics.js

```diff
@@ -35,9 +35,11 @@ const getAnalytics = async (req, res, next) => {
 
     // fetch (sample of) actions of retrieved space ids
     const spaceIds = spaceTree.map((space) => space.id);
-    const actions = await fetchActions(actionsCollection, spaceIds, {
-      sampleSize: requestedSampleSize,
-    });
+    const actions = await (
```

Comment:
I think you can just `await` on the outer function.

Reply:
I'm assuming you mean:

`const actions = await fetchActions(actionsCollection, spaceIds, {sampleSize: requestedSampleSize}).toArray();`

That doesn't work (I tested it). I think: `fetchActions` is asynchronous, so you have to `await` it; it returns the resolved value of `await collection.aggregate(...)` (a cursor), and the `.toArray()` call on that cursor is also asynchronous, so you have to `await` that. Hope I haven't missed anything.

Reply:
If you don't `return await` in line 20 of `/service/analytics` I think it should work. Can you try that?